index.js

```diff
@@ -1,102 +1,114 @@
+app.use(cookieParser())
+app.use(session({ secret: 'keyboard cat', resave: false, saveUninitialized: false }))
+auth.init(app)
 
 app.get(['/', '/signup', '/login', '/profile/:id'], (req, res) => {
-    res.sendFile(path.join(publicDir, '/index.html'));
-});
+    res.sendFile(path.join(publicDir, '/index.html'))
+})
 
 app.post('/api/login', (req, res, next) => {
     if (req.body.facebookLogin) {
         // TODO: This is completely unsecure and temporary... need to replace react-facebook-login with https://github.com/jaredhanson/passport-facebook
-        req.user = req.body;
-        return res.json(req.user);
+        req.user = req.body
+        return res.json(req.user)
     } else {
         // See: https://github.com/jaredhanson/passport-local
         passport.authenticate('local', (err, user, info) => {
             if (err || !user) {
-                console.log('error with login:', err, user);
-                return res.end();
+                console.log('error with login:', err, user)
+                return res.end()
             }
             req.login(user, () => {
-                return res.json(user);
-            });
-        })(req, res, next);
+                return res.json(user)
+            })
+        })(req, res, next)
     }
+})
+
+app.get('/api/auth/facebook', passport.authenticate('facebook'));
+
+app.get('/api/auth/facebook/callback', passport.authenticate('facebook', { failureRedirect: '/login' }), (req, res) => {
+  res.redirect(`/profile/${req.user._id}`)
 });
 
 app.get('/api/users', (req, res) => {
     if (auth.isAdmin(req)) {
         db.getAll('user').then(users => {
-            return res.json({ users });
-        });
+            return res.json({ users })
+        })
     } else {
-        return res.json({ error: 'You do not have permission to access this resource' });
+        return res.json({ error: 'You do not have permission to access this resource' })
     }
-});
+})
 
 app.get('/api/user/:id', (req, res) => {
     if (req.isAuthenticated()) {
         db.getById('user', req.params.id).then(user => {
-            return res.json({ user });
-        });
+            return res.json({ user })
+        })
     } else {
-        return res.json({ error: 'Not authenticated' });
+        return res.json({ error: 'Not authenticated' })
     }
-});
+})
 
 app.post('/api/user', (req, res) => {
-    db.insertOne('user', req.body).then(result => {
-        var userRecord = req.body;
-        userRecord.recordType = "New User";
-        mailer.notifyAdmin(userRecord);
-        return res.json(result);
+    // It is good practice to specifically pick the fields we want to insert here *in the backend*,
+    // even if we have already done so on the front end. This is to prevent malicious users
+    // from adding unexpected fields by modifying the front end JS in the browser.
+    db.insertOne('user', _.pick(req.body, [
+        'name', 'email', 'country', 'region', 'phone', 'interests', 'passphrase'
```

Comment:
since we need to pick them then we also need skills and isAdmin

Reply:
Good catch. Will add skills. I don't think we should add isAdmin otherwise anyone would be able to make themselves admin.